src/api/process_status.js

```diff
@@ -74,17 +74,62 @@ export async function processStatus(textbookRef, textbook) {
     throw new Error(`Could not find 'textbook_events' subcollection reference on textbook id: ${textbook.id}`)
   }
 
-  const textbookEvents = (await getDocs(query(textbookEventsRef, orderBy('timestamp')))).docs;
+  const textbookEvents = (await getDocs(query(textbookEventsRef, orderBy('timestamp', "desc")))).docs;
```

Comment:
If we're only using the most recent event, let's only fetch the most recent event.

If we think in the future we will change the algorithm to consider multiple events, we can leave an unused function so future developer can use that.

I think this should also save some GCP resources, IIRC it only counts returned results for cost, not if you do scanning.

Reply:
Actually this might not be true: https://firebase.google.com/docs/firestore/pricing

But regardless I think it's good practice to just retrieve the single document we need.

Reply:
yep that is fair